lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1856,6 +2000,22 @@ LogicalResult FIRRTLLowering::visitExpr(SpecialConstantOp op) {
                      getOrCreateIntConstant(APInt(/*bitWidth*/ 1, op.value())));
 }
 
+LogicalResult FIRRTLLowering::visitExpr(SubindexOp op) {
+  if (getLoweredValue(op) || !op.input())
+    return success();
```

Comment:
I think you pulled these two lines from `visitExpr(SubfieldOp` which is handling some weird mem cases specially. I don't think those apply here.  Please drop this unless you need it (and add a testcase showing that if so).

Reply:
Removed. Thanks.